### PR TITLE
gui: Add one-off notifications that need to be acked

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
...

### CODE REVIEW

**Comment by calmh** [comment]
I'm sympathetic to the concept, but I think this is a bit heavy handed. We don't want to show that notification for all new users for eternity when they install v0.13.47. Maybe if we invert the logic and keep an array of unacked notifications, and we add for example the v12v13 notification in the config migration, or for new installs of <= v0.13.8 or whatever?

**Comment by AudriusButkevicius** [comment]
That's a good point, totally missed that.

My idea was that you'd only have to specify the notifications in one place, and rest is magic.

**Comment by calmh** [comment]
Additional use cases for this that come to mind now that you mention it are pushes for "star us on github (attracts developers)!" and "consider donating?". Although we need to be very careful when rolling such things out to avoid annoying the shit out of people.

**Comment by AudriusButkevicius** [comment]
Right, new version.

File: gui/default/index.html [hunk 1]
```diff
@@ -248,6 +248,32 @@ <h3 class="panel-title">
...
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              <div class="panel-icon">
+                <span class="fa fa-exclamation-circle"></span>
+              </div>
+              <span translate>Notification</span>
+            </h3>
+          </div>
+          <div class="panel-body">
+            <p>{{ message }}</p>
```
Comment (calmh) [inline gui/default/index.html]: How does this work? Do we fling the entire message into the config, unlocalized?

Comment (AudriusButkevicius) [inline gui/default/index.html]: Yep, alternative have some magic ids or something?
Also unlikely the messages will get translated as fast as we produce them.

**Comment by calmh** [comment]
Hmm, yeah. I think your first approach there was better, with a magic ID pointing to an HTML file or whatever. Then at least there is a chance it gets translated among the other GUI strings

**Comment by AudriusButkevicius** [comment]
But this way, messages could also be handled by wrappers etc.

**Comment by AudriusButkevicius** [comment]
Bump?

**Comment by calmh** [comment]
I feel rather strongly that if we build a mechanism to show long lived notifications in the GUI, it should be translatable. Sure, there may not be time to get everything translated in time, but for things like "this release is incompatible with $foo", "we've migrated this to that" and so on we'll probably have some heads up, and it'll get translated at some point so new users will benefit in the future.

**Comment by calmh** [comment]
> My only real complaint with your first iteration was show-by-default vs add-id-to-magic-list-to-show.

Do you want to resurrect this? Otherwise I want to, because I want the feature :D

**Comment by AudriusButkevicius** [comment]
I will, just stuck with some real life shit for a past weeks.

**Comment by calmh** [comment]
Am, I'm sorry :( No rush.

**Comment by AudriusButkevicius** [comment]
Updated.

**Comment by calmh** [comment]
Sweet

**Comment by calmh** [comment]
@st-review merge it at your leasure

**Comment by st-review** [comment]
:ok_hand: Merged as 08b5a7908f4ab33ae3821caa7b9d010f616630da. Thanks, @AudriusButkevicius!

**Comment by st-review** [comment]
@calmh: Merge failed:

```
$ git fetch -f origin refs/pull/3254/head:pr-3254
$ git fetch -f origin master:orig/master
From github.com:syncthing/syncthing
   a8cd9d0..08b5a79  master     -> orig/master
$ git reset --hard
HEAD is now at 08b5a79 gui: Add one-off notifications that need to be acked
$ git checkout master
Already on 'master'
Your branch is up-to-date with 'origin/master'.
$ git reset --hard orig/master
HEAD is now at 08b5a79 gui: Add one-off notifications that need to be acked
$ git clean -fxd
Skipping repository syncthing/syncthing
Skipping repository syncthing/syncthing-android
Skipping repository syncthing/docs
Skipping repository syncthing/discosrv
Skipping repository syncthing/syncthing-gtk
Skipping repository syncthing/syncthing-inotify
$ git merge --squash --no-commit pr-3254
Automatic merge went well; stopped before committing as requested
Squash commit -- not updating HEAD
$ git commit -F -
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean

```

**Pull request closed without merging**